Carbon.FileSystem/Functions/Grant-CNtfsPermission.ps1

```diff
@@ -0,0 +1,135 @@
+function Grant-CNtfsPermission
+{
+    <#
+    .SYNOPSIS
+    Grants permission on folders and files.
+
+    .DESCRIPTION
+    The `Grant-CNtfsPermission` functions grants permissions to folders and files. Pass the folder/file path to the
+    `Path` parameter, the user/group name to the `Identity` parameter, and the permissions to the `Permission`
+    parameter. By default, the permissions are applied to the folder and inherited to all its subfolders and files. To
+    control how the permissions are applied, use the `ApplyTo` parameter. If you want permissions to only apply to child
+    files and folders, use the `OnlyApplyToChildFilesAndFolders` switch.
+
+    By default, an "Allow" permission is granted. To add a "Deny" permission, set the value of the `Type` parameter to
+    `Deny`.
+
+    All existing, non-inherited permissions for the given identity are removed first. If you want to preserve a
+    user/group's existing permissions, use the `Append` switch.
+
+    To remove *all* non-inherited permissions except the permission being granted, use the `Clear` switch.
+
+    The permission is only granted if it doesn't exist. To always grant the permission, use the `Force` switch.

```

Comment:
I'd rephrase from `To always grant the permission` to `To always apply the permission` since the function can add Deny permissions (i.e. not a "grant").

Reply:
That's the way NTFS permissions are designed: there are allow rules and deny rules, but they both get granted the same way. It's a permission type, not a permission.